tests/conversion

```diff
@@ -0,0 +1,264 @@
+    else
+        echo "===> FAIL: IPv4 address: ${instName}"
+        fail=1
+    fi
+
+    # IPv6 address
+    if echo "${address}" | grep ":" -q; then
+        echo "===> PASS: IPv6 address: ${instName}"
+    else
+        echo "===> FAIL: IPv6 address: ${instName}"
+        fail=1
+    fi
+
+    return "${fail}"
+}
+
+checkDNS() {
+    instName=$1
+
+    # Test lxd-agent and DNS resolution.
+    if lxc exec "${instName}" -- nslookup canonical.com >/dev/null 2>&1; then
+        echo "===> PASS: DNS resolution: ${instName}"
+        return 0
+    fi
+
+    if lxc exec "${instName}" -- getent hosts canonical.com >/dev/null 2>&1; then
+        echo "===> PASS: DNS resolution: ${instName}"
+        return 0
+    fi
+
+    echo "===> FAIL: DNS resolution: ${instName}"
+    return 1
+}
+
+# conversion_vm creates a new LXD virtual machine from the image on the given path.
+# A dedicated storage pool of the given type will be created for a new virtual machine.
+#
+# Input arguments:
+# 1: Name of new virtual machine.
+# 2: Type of the storage pool.
+# 3: Path to the image.
+conversion_vm() {
+    conversion vm "$1" "$2" "$3"
+}
+
+# conversion_container creates a new LXD container from the filesystem of the existing container.
+#
+# Input arguments:
+# 1: Name of the existing container.
+conversion_container() {
+    conversion container "$1" "" "/"
+}
+
+# conversion runs lxd-migrate for the given image path. For virtual machine, lxd-migrate is
+# executed on the localhost. For container, lxd-migrate will be installed and executed from
+# an existing container.
+conversion() {
+    instType=$1
+    instName=$2
+    poolType=$3
+    imgPath=$4
+
+    if [ "${instType}" = "vm" ]; then
+        echo "==> TEST: Conversion: Import virtual-machine '${instName}' on '${poolType}' storage pool"
+
+        poolName="conversion-${poolType}"
+        hostAddr="127.0.0.1"
+        instTypeCode="2" # VM in migration questioneer.
+
+        lxdMigrateCmd="lxd-migrate --conversion format"
+
+        # Ensure instance and storage pool with the given name do not exist.
+        lxc delete -f "${instName}" 2>/dev/null || true
+        lxc storage delete "${poolName}" 2>/dev/null || true
+
+        # Create storage pool.
+        lxc storage create "${poolName}" "${poolType}"
+
+    elif [ "${instType}" = "container" ]; then
+        echo "==> TEST: Conversion: Import container '${instName}'"
+
+        poolName=""
+        hostAddr=$(lxc network show lxdbr0 | awk '/ipv4.address:/ {split($2, arr, "/"); print arr[1]}')
+        instTypeCode="1" # Container.
+
+        lxdMigrateCmd="lxc exec ${instName} -- /root/go/bin/lxd-migrate"
+
+        # Install rsync and lxd-migrate.
+        lxc exec "${instName}" -- apt-get update
+        lxc exec "${instName}" -- apt-get install rsync file
+        lxc exec "${instName}" -- snap install go --channel=latest/edge --classic
+        lxc exec "${instName}" -- go install github.com/canonical/lxd/lxd-migrate@latest
```

Comment:
No `CGO_ENABLED=0` needed there?

Reply:
Didn't even realize that it is missing here. However, it complains on the runner if CGO is not disabled

Reply:
@MusicDin is this addressed?

Reply:
Should this be?

```
lxc exec "${instName}" --env=CGO_ENABLED=0 -- go install github.com/canonical/lxd/lxd-migrate@latest
```

Reply:
Fixed.